### PR TITLE
rp2040: rename all usages of pwm channel to pwm slices

### PR DESCRIPTION
**Pull request by alexandruradovici**
This pull request renames all the PWM data structures of RP2040 from `Channel` to `Slice`. It modifies the comments to reflect the renames. It follows #2762.

### CODE REVIEW

File: embassy-rp/src/pwm.rs [hunk 1]
```diff
@@ -301,78 +301,78 @@ impl<'d, T: Channel> Drop for Pwm<'d, T> {
...
+slice!(PWM_SLICE1, 1);
+slice!(PWM_SLICE2, 2);
+slice!(PWM_SLICE3, 3);
+slice!(PWM_SLICE4, 4);
+slice!(PWM_SLICE5, 5);
+slice!(PWM_SLICE6, 6);
+slice!(PWM_SLICE7, 7);
+
+/// PWM Channel A.
+pub trait PwmChannelA<T: Slice>: GpioPin {}
```
Comment (Dirbaio) [inline embassy-rp/src/pwm.rs:339]: I'd prefer to keep the pin traits as `PwmPin*`, for consistency with other drivers (like TxPin, SdaPin...). These "pin traits" indicate "this pin is usable for this function", it makes sense to call them all "*Pin".

Comment (Dirbaio) [reply embassy-rp/src/pwm.rs:339]: Or perhaps we can name them `ChannelAPin`, `ChannelBPin`. as in "pin usable for channel A". Consistent with "TxPin" as in "Pin usable for TX".

Comment (alexandruradovici) [reply embassy-rp/src/pwm.rs:339]: I think `ChannelAPin` and `ChannelBPin` work very nice.

**Review by Dirbaio — APPROVED** [review]
thanks!

**Comment by Dirbaio** [comment]
as a side note: it's confusing/inconsistent that the PAC calls the slices "ch"! I think that's where the "wrong" name in embassy-rp originally came from. And that inconsistency is straight from Raspberry Pi... :smiling_face_with_tear:

**Comment by alexandruradovici** [comment]
> as a side note: it's confusing/inconsistent that the PAC calls the slices "ch"! I think that's where the "wrong" name in embassy-rp originally came from. And that inconsistency is straight from Raspberry Pi... 🥲

I figured this out while renaming the pins and verified the data sheet. They call the PWM slices, but the registers are `CH0` ...

**Comment by Dirbaio** [comment]
¯\_(ツ)_/¯

**Pull request merged**